syft/file/rekor_cataloger.go

```diff
@@ -30,26 +31,25 @@ func (c *Cataloger) UsesExternalSources() bool {
 	return true
 }
 
-func (c *Cataloger) Catalog(resolver source.FileResolver) ([]pkg.Package, []artifact.Relationship, error) {
+func (c *Cataloger) Catalog(resolver source.FileResolver) ([]artifact.Relationship, error) {
```

Comment:
design: The return signature here is interesting and I think I need to explore the file package code a bit more before giving good feedback here.

I looked at the other catalogers in this package:
- classification
- contents
- digest
- metadata
- secrets

Each of those returns something akin to `map[source.Coordinates]someValue`. Even though this is not enforced by an interface implementation at a higher level, do we want the `rekor_cataloger` to be different in this case and return `[]Relationship` as the output for `Catalog`?

Personally, I think this is ok since we're describing a union to an external source and we're doing all of the work upfronts in the cataloger of creating the external ref and relationships without additional overhead. I just wanted to point out that this cataloger was a little different from its current sibblings.

Reply:
I agree that this is ok. I think that the ability to implement catalogers with different return types is one of the benefits of implementing a cataloger at the task level as opposed to the package-cataloger level. `syft.CatalogPackages` in `generateCatalogPackagesTask` also has a different return type.